packages/model-viewer/src/three-components/ARRenderer.ts

```diff
@@ -256,6 +292,84 @@ export class ARRenderer extends EventDispatcher<
     this.dispatchEvent({type: 'status', status: ARStatus.SESSION_STARTED});
   }
 
+  private intersectObjects(controller: any) {
+    // Do not highlight in mobile-ar
+    if (controller.userData.targetRayMode === 'screen') return;
+
+    // Do not highlight when already selected
+    if (controller.userData.selected !== undefined) return;
+
+    const line = controller.getObjectByName('line');
+    const intersections = this.getIntersections(controller);
+
+    if (intersections.length > 0) {
+      const intersection = intersections[0];
+
+      const object = intersection.object;
+      // @ts-ignore
+      object.material.emissive.setHex(0x333333);
+
+      this.intersected.push(object);
+
+      line.scale.z = intersection.distance;
+    } else {
+      line.scale.z = 5;
+    }
+  }
+
+  private getIntersections(controller: any) {
+    controller.updateMatrixWorld();
+
+    const tempMatrix = new Matrix4();
+
+    tempMatrix.identity().extractRotation(controller.matrixWorld);
+
+    const raycaster: Raycaster = new Raycaster();
+
+    raycaster.ray.origin.setFromMatrixPosition(controller.matrixWorld);
+    raycaster.ray.direction.set(0, 0, -1).applyMatrix4(tempMatrix);
+
+    const group = this.presentedScene!.model;
+    // @ts-ignore
```

Comment:
Can you try to replace these `@ts-ignore`s by casting, e.g. `const group = this.presentedScene!.model as Group;` or something? That's a lot more specific and helps me to understand the problem.

Reply:
done.